src/modules/home/routes/single-property.route.tsx

```diff
@@ -9,10 +9,38 @@ import { singlePropertyStyles } from './single-property.route.styles';
 import { Property } from '../components/single-property/single-property-amenity';
 import { PropertyReview } from '../components/single-property/single-property-review';
 import Divider from '@mui/material/Divider';
-import { accommodation } from '../temp-data/single-accommodation';
+import { useAccommodationById } from '../api/get-accommodation';
+import { useParams } from 'react-router-dom';
+import { type Accommodation } from '../types/accommodation.type';
 
 export function SinglePropertyRoute() {
   const isMobile = useMediaQuery('(max-width:700px)');
+  const { id } = useParams();
+
+  const { data, error, isLoading } = useAccommodationById(id || '');
+
+  if (!id) {
+    return <div>Error: Property ID is not provided.</div>;
+  }
+
+  if (isLoading) return <div>Loading...</div>;
+  if (!data) return <div>No data found</div>;
+  if (error instanceof Error) return <div>Error: {error.message}</div>;
+
+  const accommodationData: Accommodation = data.data;
+
+  const reviews = accommodationData.reviews.map(
+    (review: { id: string; user: { firstName: string; lastName: string }; comment: string; rating: number }) => ({
```

Comment:
I think `Pick<User, "firstName" | "lastName">` would be a bit clearer

Reply:
If it (user info) is specific only to an accommodation, then it's not really necessary. But if it is the same for other entities like reviews, then yes, a separate type is much more beneficial.

Reply:
it is used for only accommodation